File: app/src/main/java/pro/anton/averin/networking/testrest/presenters/AddHeaderPopupPresenter.java
package pro.anton.averin.networking.testrest.presenters;

import java.util.List;

import javax.inject.Inject;
import javax.inject.Singleton;

import pro.anton.averin.networking.testrest.data.Repository;
import pro.anton.averin.networking.testrest.data.models.Headers;

@Singleton
public class AddHeaderPopupPresenter extends BasePresenter<AddHeaderPopupView> {

    @Inject
    Repository repository;

    @Inject
    public AddHeaderPopupPresenter() {
    }

    public List<Headers.Header> getSupportedHeaders() {
        return repository.getSupportedHeaders();
    }

    public boolean updateHeader(Headers.Header selectedHeader) {
        return repository.updateHeader(selectedHeader);
    }
}

File: app/src/main/java/pro/anton/averin/networking/testrest/presenters/AddHeaderPopupPresenter.kt
package pro.anton.averin.networking.testrest.presenters

import pro.anton.averin.networking.testrest.data.Repository
import pro.anton.averin.networking.testrest.data.models.Headers
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class AddHeaderPopupPresenter
@Inject
constructor() : BasePresenter<AddHeaderPopupView>() {

    @Inject
    lateinit var repository: Repository

    val supportedHeaders: List<Headers.Header>
        get() = repository.getSupportedHeaders()

    fun updateHeader(selectedHeader: Headers.Header): Boolean {
        return repository.updateHeader(selectedHeader)
    }
}
